### PR TITLE
Unit Tests for SideNav

### PR DESCRIPTION
**Pull request by StevenUlmer**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/mobi/goponents/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
~~Docs have been added / updated (for bug fixes / features)~~

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [x] Other... Please describe: Unit Tests

## What is the current behavior?
Issue Number: https://github.com/mobi/goponents/issues/385

## What is the new behavior?
There are unit tests for the sidenav components and service

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

### CODE REVIEW

File: projects/go-lib/src/lib/components/go-side-nav/go-nav-item/go-nav-item.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      routeTitle: 'Help'
+    };
+
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+});
```
Comment (grahamhency) [inline projects/go-lib/src/lib/components/go-side-nav/go-nav-item/go-nav-item.component.spec.ts]: Should we also test the functionality in `ngOnInit`?

File: projects/go-lib/src/lib/components/go-side-nav/go-side-nav/go-side-nav.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+      nestedGroup.expanded = true;
+
+      component.closeNavs(component.menuItems[0] as NavGroup);
+
+      expect(nestedGroup.expanded).toBe(false);
+      expect(component.menuItems[1]['expanded']).toBe(false);
+      expect(component.menuItems[0]['expanded']).toBe(true);
+    });
+  });
+});
```
Comment (grahamhency) [inline projects/go-lib/src/lib/components/go-side-nav/go-side-nav/go-side-nav.component.spec.ts]: Should we test the other methods in this file as well? I know they're private, but we can test them still.

File: projects/go-lib/src/lib/components/go-side-nav/go-side-nav/go-side-nav.service.spec.ts [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    });
+
+    service = TestBed.get(GoSideNavService);
+  });
+
+  it('should create', () => {
+    expect(service).toBeTruthy();
+  });
+
+  describe('setMenuItems', () => {
```
Comment (grahamhency) [inline projects/go-lib/src/lib/components/go-side-nav/go-side-nav/go-side-nav.service.spec.ts]: Should we also test that `createNavMap` get's called?

File: projects/go-lib/src/lib/components/go-side-nav/go-side-nav/go-side-nav.service.spec.ts [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    it('should set previously active items to false', () => {
+      service.setMenuItems(navItems);
+      service.setActiveItem('/group_item_1');
+      service.setActiveItem('/level_1_item');
+
+      expect(service.menuItems[0].routeActive).toBe(false);
+      expect(service.menuItems[1].routeActive).toBe(true);
+    });
+  });
+});
```
Comment (grahamhency) [inline projects/go-lib/src/lib/components/go-side-nav/go-side-nav/go-side-nav.service.spec.ts]: Should we test the private methods in this service as well?

File: projects/go-lib/src/lib/components/go-side-nav/go-side-nav/go-side-nav.component.spec.ts (hunk 2 above)
Comment (StevenUlmer) [reply projects/go-lib/src/lib/components/go-side-nav/go-side-nav/go-side-nav.component.spec.ts]: So here's my thoughts on this. I think as long as we test everything in the public methods that we actually want to happen, there's not really a reason to test the private methods. We don't really care how the menuItems gets expanded, we just care that when we click on an item it gets set to expanded. Private methods are basically just helpers and ways to not duplicate code. If we write good tests around the public methods that test everything and then also test the private methods it creates duplicate tests. I think this response on stack overflow has really good insight on testing private methods (https://stackoverflow.com/questions/105007/should-i-test-private-methods-or-only-public-ones/47401015#47401015). Thoughts?

Comment (grahamhency) [reply projects/go-lib/src/lib/components/go-side-nav/go-side-nav/go-side-nav.component.spec.ts]: @StevenUlmer yeah that's fine. I forgot we already talked and decided on this. Disregard my comments on them.
